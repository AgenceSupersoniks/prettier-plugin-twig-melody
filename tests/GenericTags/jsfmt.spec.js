run_spec(__dirname, ["melody"], {
    twigMultiTags: [
        "nav,endnav",
        "switch,case,endswitch",
        "ifchildren,endifchildren",
        "cache,endcache"
    ]
});
