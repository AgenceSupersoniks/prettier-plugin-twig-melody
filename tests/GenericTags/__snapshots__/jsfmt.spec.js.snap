// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cache.melody.twig 1`] = `
{% cache globally using key craft.some.rather.long.property.chain.request.path for 3 weeks %}
    {% for block in entry.myMatrixField %}
        <p>{{ block.text }}</p>
    {% endfor %}
{% endcache %}

{# prettier-ignore #}
{% cache globally using key craft.some.rather.long.property.chain.request.path for 3 weeks %}
    {% for block in entry.myMatrixField %}
        <p>{{ block.text }}</p>
    {% endfor %}
{% endcache %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% cache globally

    using
    key
    craft.some.rather.long.property.chain.request.path
    for
%}
    {% for block in entry.myMatrixField %}
        <p>
            {{ block.text }}
        </p>
    {% endfor %}
{% endcache %}

{# prettier-ignore #}
{% cache globally using key craft.some.rather.long.property.chain.request.path for 3 weeks %}
    {% for block in entry.myMatrixField %}
        <p>{{ block.text }}</p>
    {% endfor %}
{% endcache %}

`;

exports[`header.melody.twig 1`] = `
{% header "Cache-Control: max-age=" ~ (expiry.timestamp - now.timestamp) %}

{# prettier-ignore #}
{% header "Cache-Control: max-age=" ~ (expiry.timestamp - now.timestamp) %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% header 'Cache-Control: max-age=' ~ expiry.timestamp - now.timestamp %}

{# prettier-ignore #}
{% header "Cache-Control: max-age=" ~ (expiry.timestamp - now.timestamp) %}

`;

exports[`includeCssFile.melody.twig 1`] = `
{% includeCssFile "/assets/css/layouts/" ~ entry.layout ~ ".css" %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% includeCssFile '/assets/css/layouts/' ~ entry.layout ~ '.css' %}

`;

exports[`nav.melody.twig 1`] = `
{% nav entry in entries %}
    <li>
        <a href="{{ entry.url }}">{{ entry.title }}</a>
        {% ifchildren %}
            <ul>
                {% children %}
            </ul>
        {% endifchildren %}
    </li>
{% endnav %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% nav entry in entries %}
    <li>
        <a href="{{ entry.url }}">{{ entry.title }}</a>
        {% ifchildren %}
            <ul>
                {% children %}
            </ul>
        {% endifchildren %}
    </li>
{% endnav %}

`;

exports[`paginate.melody.twig 1`] = `
{% paginate craft.entries.section('blog').limit(10) as pageInfo, pageEntries %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% paginate craft.entries.section('blog').limit(10)

    as
    pageInfo
    ,
    pageEntries
%}

`;
